### PR TITLE
Enginify `trans_rect`

### PR DESCRIPTION
**Pull request by glebm**
Moves `trans_rect` to `engine` as `DrawHalfTransparentRectTo` and makes it operate on `CelOutputBuf`.

### CODE REVIEW

File: Source/engine.cpp [hunk 1]
```diff
@@ -555,6 +555,32 @@ void DrawLineTo(CelOutputBuffer out, int x0, int y0, int x1, int y1, BYTE color_
 	}
 }
 
+void DrawHalfTransparentRectTo(CelOutputBuffer out, int sx, int sy, int width, int height)
+{
+	int row, col;
+	BYTE *pix = out.at(sx, sy);
+
+	if (sgOptions.bBlendedTransparancy) { // Blended
```
Comment (AJenbo) [inline Source/engine.cpp:563]: maybe we should split this in to two function?

**Review by AJenbo — APPROVED** [review]

File: Source/engine.cpp (hunk 1 above)
Comment (glebm) [reply Source/engine.cpp]: Done

File: Source/engine.h [hunk 2]
```diff
@@ -340,6 +340,23 @@ void Cl2DrawLight(int sx, int sy, BYTE *pCelBuff, int nCel, int nWidth);
...
 void DrawLineTo(CelOutputBuffer out, int x0, int y0, int x1, int y1, BYTE color_index);
 
+
+/**
+ * Draws a half-transparent rectangle by blacking out odd pixels on odd lines,
+ * even pixels on even lines.
+ *
+ * If blended transparency is enabled, uses `paletteTransparencyLookup` table instead
+ * of blacking pixels out.
+ *
```
Comment (AJenbo) [inline Source/engine.h:350]: This is implementation-specific and global transparency handling is described elsewhere already.
```suggestion
 *
```

Comment (AJenbo) [inline Source/engine.h:350]: This is implementation-specific and global transparency handling is described elsewhere already.
```suggestion
 *
```

**Pull request merged**